packages/steiger/src/models/config/convert-relative-globs-to-absolute.ts

```diff
@@ -0,0 +1,41 @@
+import { posix, sep } from 'node:path'
+import { Config } from '@steiger/types'
+
+import { isConfigObject, isConfiguration } from './raw-config'
+
+function convertGlobs(rootPath: string, globs: Array<string>) {
+  function composeAbsolutePath(root: string, glob: string) {
+    // Remove '/'. The root has platform-specific separators
+    const segmentsOfRoot = root.slice(1).split(sep)
+
+    return `/${posix.join(...segmentsOfRoot, glob)}`
+  }
+
+  return globs.map((glob) => (glob.startsWith('.') ? composeAbsolutePath(rootPath, glob) : glob))
+}
+
+export function convertRelativeGlobsToAbsolute(config: Config, configLocationPath: string | null) {
```

Comment:
praise: great that you decided to extract it into a separate module! It's easier to maintain and test that way